shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -202,10 +202,10 @@ final class TsTypegenCodeBuilder {
         val fieldTypesCode = if (fieldTypes.length === 1) {
```

Comment:
`distinct` may return a list with only one element. So I'd rather have this be:

```suggestion
        val fieldTypesCode = fieldTypes. distinct match {
          case fieldType :: Nil => ...
          case fieldTypes => ...
        }
```